chain/tests/mine_simple_chain.rs

```diff
@@ -60,14 +61,52 @@ fn setup(dir_name: &str, genesis: Block) -> Chain {
 #[test]
 fn mine_empty_chain() {
 	global::set_mining_mode(ChainTypes::AutomatedTesting);
-	let chain = setup(".grin", pow::mine_genesis_block().unwrap());
-	let keychain = ExtKeychain::from_random_seed().unwrap();
+	let keychain = keychain::ExtKeychain::from_random_seed().unwrap();
+	mine_some_on_top(".grin", pow::mine_genesis_block().unwrap(), &keychain);
+}
+
+#[test]
+fn mine_genesis_reward_chain() {
+	global::set_mining_mode(ChainTypes::AutomatedTesting);
+
+	// add coinbase data from the dev genesis block
+	let mut genesis = genesis::genesis_dev();
+	let keychain = keychain::ExtKeychain::from_random_seed().unwrap();
+	let key_id = keychain::ExtKeychain::derive_key_id(0, 1, 0, 0, 0);
+	let reward = reward::output(&keychain, &key_id, 0, 0).unwrap();
+	genesis.body.outputs.push(reward.0);
```

Comment:
Noticed this and wondered if we could do something similar to the "factory methods" on transaction - 

https://github.com/mimblewimble/grin/blob/8d8f533b8e8d8df77ddb800f7f8366ada4b617fc/core/src/core/transaction.rs#L798-L806

That way we could avoid making `body` public.

That said maybe we don't want to worry too much about it right now as we are still iterating on exactly how this genesis block is going to be constructed in mainnet. 

We haven't really had a need to add arbitrary outputs/kernels to a block before. This will likely be clearer later.

Reply:
Great idea. I've added a `with_reward` for that purpose. Not that there was already one that was more of a constructor, and renamed it `from_reward` which I think conveys what it does better. Perhaps we should try to use `from_` everywhere when the semantic is closer to a `new`.